src/cargo/core/compiler/fingerprint/mod.rs

```diff
@@ -1865,16 +1988,32 @@ pub fn parse_dep_info(
     };
     let mut ret = RustcDepInfo::default();
     ret.env = info.env;
-    ret.files.extend(info.files.into_iter().map(|(ty, path)| {
-        match ty {
-            DepInfoPathType::PackageRootRelative => pkg_root.join(path),
-            // N.B. path might be absolute here in which case the join will have no effect
-            DepInfoPathType::TargetRootRelative => target_root.join(path),
-        }
-    }));
+    ret.files.extend(
+        info.files
+            .into_iter()
+            .map(|(ty, path)| make_absolute_path(ty, pkg_root, path, target_root)),
+    );
+    for (ty, path, file_len, checksum) in info.checksum {
+        let path = make_absolute_path(ty, pkg_root, path, target_root);
+        ret.checksum
+            .insert(path, (file_len, Checksum::from_str(&checksum)?));
+    }
     Ok(Some(ret))
 }
 
+fn make_absolute_path(
+    ty: DepInfoPathType,
+    pkg_root: &Path,
+    path: PathBuf,
+    target_root: &Path,
```

Comment:
nit: I'd make `path` the last parameter.  The first three are all related in determining what the `root` should be

Alternatively, you could just make this a `path_root` function that only takes the first three and the join is done in the caller

Reply:
Resolved in 5afdc115d295f74c475a2252df40ff7d2351c3eb